spark/v3.2/spark/src/test/java/org/apache/iceberg/spark/actions/TestRewriteDataFilesAction.java

```diff
@@ -1038,6 +1038,76 @@ public void testInvalidAPIUsage() {
         "Cannot set strategy", () -> actions().rewriteDataFiles(table).sort(SortOrder.unsorted()).binPack());
   }
 
+  @Test
+  public void testRewriteJobOrderBytes() {
+    Table table = createTablePartitioned(4, 2);
```

Comment:
it would be great if these tests checked to see if the ordering actually worked, I think this may be possible through some package private methods which the test class is allowed to access?

Reply:
I agree, I will add the test cases in the next commit.

Reply:
I have changed `planFileGroups()` and `toGroupStream()` to package private methods and have added the tests which checks if the ordering actually worked.